src/components/floating-action-buttons/floating-action-buttons.component.ts

```diff
@@ -0,0 +1,41 @@
+import { Component, Input, ChangeDetectionStrategy, HostListener, ElementRef } from '@angular/core';
+import { trigger, state, style, transition, animate, query, stagger } from '@angular/animations';
+
+import { FloatingActionButtonComponent } from './floating-action-button.component';
+import { FloatingActionButtonsService } from './floating-action-buttons.service';
+
+@Component({
+    selector: 'ux-floating-action-buttons',
+    templateUrl: './floating-action-buttons.component.html',
+    providers: [FloatingActionButtonsService],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    preserveWhitespaces: false,
+    animations: [
+        trigger('fabAnimation', [
```

Comment:
The main concern I have with this is whether we can override the animations in the quantum stylesheet for Micro Focus (my understanding is that we cannot).

Reply:
You are correct, we cannot override it by using the stylesheets. I have proposed a solution in my latest changes, if you like it I can create a jira to update our docs/seed project with how to use it, if not I can revert to CSS animations but we will lose the stagger in the button animations. 

There are cases where it would be good for our components to know what the active theme is, this would allow us to show and hide certain elements based on the theme or use different animations etc..

Currently teams set the theme by injecting the colorService eg:

```
constructor(colorService: ColorService) {
   colorService.setColorSet(colorSets.microFocus);
}
```

I have created a ThemeService that would allow the teams to do something similar:

```
constructor(themeService: ThemeService) {
   themService.setTheme(Theme.MicroFocus);
}
```

Upon setting the theme this would automatically update color palette and allow any components to subscribe to an active theme and update anything based on the active theme and use the appropriate animations etc..

(I also have it so even if they are setting the theme only by the colorService it will update the active theme based on the color set provided to prevent any issues with teams not updating to use the theme service)

Reply:
I totally missed this explanation, sorry.

Reply:
Let's make a ticket on this topic and discuss it during storytime. In the meantime we can keep these animations so that it is consistent with the ng1 buttons.